snapshot-jdbc-test/src/main/java/com/hazelcast/jet/tests/snapshot/jdbc/JdbcSinkTest.java

```diff
@@ -0,0 +1,165 @@
+    private static final String DATABASE_NAME = "snapshot-jdbc-test-db";
+
+    private static final String DEFAULT_DATABASE_URL = "jdbc:mysql://localhost";
+    private static final int DEFAULT_SLEEP_MS_BETWEEN_ITEM = 50;
+    private static final int DEFAULT_SNAPSHOT_INTERVAL = 5000;
+
+    private String connectionUrl;
+
+    private int sleepMsBetweenItem;
+    private long snapshotIntervalMs;
+
+    public static void main(String[] args) throws Exception {
+        new JdbcSinkTest().run(args);
+    }
+
+    @Override
+    public void init(JetInstance client) throws Exception {
+        connectionUrl = System.getProperty("connectionUrl", DEFAULT_DATABASE_URL)
+                + "/" + DATABASE_NAME + "?useSSL=false";
+
+        sleepMsBetweenItem = propertyInt("sleepMsBetweenItem", DEFAULT_SLEEP_MS_BETWEEN_ITEM);
+        snapshotIntervalMs = propertyInt("snapshotIntervalMs", DEFAULT_SNAPSHOT_INTERVAL);
+    }
+
+    @Override
+    protected boolean runOnBothClusters() {
+        return true;
+    }
+
+    @Override
+    public void test(JetInstance client, String name) throws Exception {
+        String tableName = (TABLE_PREFIX + name).replaceAll("-", "_");
+        try (Connection connection = ((DataSource) getDataSourceSupplier(connectionUrl).get()).getConnection()) {
+            connection.createStatement().execute("DROP TABLE IF EXISTS " + tableName);
+            connection.createStatement().execute("CREATE TABLE " + tableName
+                    + "(id int PRIMARY KEY AUTO_INCREMENT, value int)");
+        }
+        JdbcSinkVerifier verifier = new JdbcSinkVerifier(name, logger, connectionUrl);
+        verifier.start();
+
+        try {
+            JobConfig jobConfig = new JobConfig();
+            jobConfig.setName(name);
+            jobConfig.setSnapshotIntervalMillis(snapshotIntervalMs);
+            jobConfig.setProcessingGuarantee(ProcessingGuarantee.EXACTLY_ONCE);
+            if (name.startsWith(STABLE_CLUSTER)) {
+                jobConfig.addClass(JdbcSinkTest.class, JdbcSinkVerifier.class);
+            }
+            Job job = client.newJob(pipeline(tableName), jobConfig);
+
+            try {
+                long begin = System.currentTimeMillis();
+                while (System.currentTimeMillis() - begin < durationInMillis) {
+                    if (getJobStatusWithRetry(job) == FAILED) {
+                        job.join();
+                    }
+                    verifier.checkStatus();
+                    sleepMinutes(1);
+                }
+            } finally {
+                job.cancel();
+            }
+        } finally {
+            verifier.finish();
+        }
+    }
+
+    @Override
+    protected void teardown(Throwable t) throws Exception {
+    }
+
+    private Pipeline pipeline(String tableName) {
+        int sleep = sleepMsBetweenItem;
+
+        Pipeline pipeline = Pipeline.create();
+
+        StreamSource<Integer> source = SourceBuilder
```

Comment:
we use this source for other tests too, time to move it to a central place I guess. can be done as a separate PR